### PR TITLE
✨ Is686/webserver api `0.12`: project ports metadata

### PR DESCRIPTION
**Pull request by pcrespov**
## What do these changes do?

- webserver OAS ``0.11.0 → 0.12.0``
   - exposes  ``GET /projects/*/metadata/ports -> Envelope[list[ProjectMetadataPortGet]]``
   image

Next PR #3623 will connect with the api-server

## Related issue/s

- Part of ITISFoundation/osparc-issues#686

## How to test

```
cd services/web/server
make install-dev
pytest -vv tests/**/test_*port*.py
```

## Checklist

- [x] ``make version-*``
- [x] ``make openapi.json``
- [x] Unit tests for the changes exist
- [x] Runs in the swarm
- [x] Documentation reflects the changes

### CODE REVIEW

File: api/specs/webserver/openapi-projects-ports.yaml [hunk 1]
```diff
@@ -20,7 +20,7 @@ paths:
           content:
             application/json:
               schema:
-                $ref: '#/components/schemas/Envelope_dict_uuid.UUID__simcore_service_webserver.projects.projects_ports_handlers.ProjectPortGet__'
+                $ref: '#/components/schemas/Envelope_dict_uuid.UUID__simcore_service_webserver.projects.projects_ports_handlers.ProjectInputGet__'
```
Comment (sanderegg) [inline api/specs/webserver/openapi-projects-ports.yaml:23]: Should not be ProjectInput**s**Get ?

File: api/specs/webserver/openapi-projects-ports.yaml [hunk 2]
```diff
@@ -42,15 +42,15 @@ paths:
               title: Updates
               type: array
               items:
-                $ref: '#/components/schemas/ProjectPort'
+                $ref: '#/components/schemas/ProjectInputUpdate'
```
Comment (sanderegg) [inline api/specs/webserver/openapi-projects-ports.yaml:45]: same here?

File: api/specs/webserver/openapi-projects-ports.yaml [hunk 3]
```diff
@@ -72,18 +72,62 @@ paths:
           content:
             application/json:
               schema:
-                $ref: '#/components/schemas/Envelope_dict_uuid.UUID__simcore_service_webserver.projects.projects_ports_handlers.ProjectPortGet__'
+                $ref: '#/components/schemas/Envelope_dict_uuid.UUID__simcore_service_webserver.projects.projects_ports_handlers.ProjectOutputGet__'
```
Comment (sanderegg) [inline api/specs/webserver/openapi-projects-ports.yaml:75]: Same here?

File: api/specs/webserver/openapi-projects-ports.yaml [hunk 4]
```diff
@@ -72,18 +72,62 @@ paths:
...
+          format: uuid
+        name: project_id
+        in: path
+      responses:
+        '200':
+          description: Successful Response
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Envelope_list_simcore_service_webserver.projects.projects_ports_handlers.ProjectMetadataPortGet__'
```
Comment (sanderegg) [inline api/specs/webserver/openapi-projects-ports.yaml:97]: ProjectMetadataPort**s**Get ?

Comment (sanderegg) [reply api/specs/webserver/openapi-projects-ports.yaml:97]: not sure I get it. is that one port or multiple port**s** ?

File: api/specs/webserver/openapi-projects-ports.yaml (hunk 1 above)
Comment (sanderegg) [reply api/specs/webserver/openapi-projects-ports.yaml:23]: I'm confused with your usage of singular vs plurals here

**Review by sanderegg — APPROVED** [review]
Great work! Please check the openapi with regard to singular vs plural.. this looks confusing or maybe it's not my day...

File: api/specs/webserver/openapi-projects-ports.yaml (hunk 4 above)
Comment (pcrespov) [reply api/specs/webserver/openapi-projects-ports.yaml:97]: yes! My fault

File: api/specs/webserver/openapi-projects-ports.yaml (hunk 1 above)
Comment (pcrespov) [reply api/specs/webserver/openapi-projects-ports.yaml:23]: aha, ok. you got me confused for a moment but i think it is correct.  The reponse body is a ``list[ProjectInputGet]`` therefore it refers to a single item. See the signatures of the routes in 
https://github.com/ITISFoundation/osparc-simcore/blob/a7bd450846fc7c770067ac424cc020e40412aa94/api/specs/webserver/scripts/openapi_projects_ports.py#L33

There is more clear. The entire OAS is auto-generated with those

File: api/specs/webserver/openapi-projects-ports.yaml (hunk 2 above)
Comment (pcrespov) [reply api/specs/webserver/openapi-projects-ports.yaml:45]: same as above ``updates: list[ProjectInputUpdate]``

File: api/specs/webserver/openapi-projects-ports.yaml (hunk 4 above)
Comment (pcrespov) [reply api/specs/webserver/openapi-projects-ports.yaml:97]: the name is auto-generated by the code in ``openapi_projects_ports.py``
The response model is ``response_model=Envelope[list[ProjectMetadataPortGet]]``

File: api/specs/webserver/openapi-projects-ports.yaml (hunk 3 above)
Comment (pcrespov) [reply api/specs/webserver/openapi-projects-ports.yaml:75]: same as above

**Review by GitHK — APPROVED** [review]
👍 think I've added all my comments on the previous PR

**Pull request merged**